device/iot-device-samples/device-reconnection-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/Application.java

```diff
@@ -0,0 +1,168 @@
+    private static final int SLEEP_DURATION_IN_SECS = 60;
+    private static AtomicBoolean stopRunning = new AtomicBoolean(false);
+
+    static boolean reconnectIndefinitely = true;
+    static List<String> failedMessageListOnClose = new ArrayList<>(); // List of messages that failed on close
+    private static CountDownLatch countDownLatch = new CountDownLatch(2);
+    private static DeviceClientManager deviceClientManager;
+
+    /**
+     * Sends a number of messages to an IoT. Default protocol is to use AMQP transport.
+     *
+     */
+    public static void main(String[] args)
+            throws IOException, URISyntaxException, InterruptedException {
+        log.debug("Starting the sample...");
+        log.debug("Using communication protocol: {}", protocol.name());
+
+        DeviceClient client = new DeviceClient(deviceConnectionString, protocol);
+        log.debug("Successfully created an IoT Hub client.");
+
+        deviceClientManager = new DeviceClientManager(client);
+        deviceClientManager.setOperationTimeout(DEVICE_OPERATION_TIMEOUT_IN_MINUTES);
+
+        deviceClientManager.connect();
+        log.debug("Opened connection to IoT Hub.");
+
+        log.debug("Setting C2D message handler...");
+        deviceClientManager.setMessageCallback(new MessageCallback(), new Object());
+
+        log.debug("Setting method handler...");
+        deviceClientManager.subscribeToDeviceMethod(new SampleDeviceMethodCallback(), null, new DeviceMethodStatusCallBack(), null);
+
+        log.debug("Start device Twin and get remaining properties...");
+        // Properties already set in the Service will shows up in the generic onProperty callback, with value and version.
+        Succeed.set(false);
+        client.startDeviceTwin(new DeviceTwinStatusCallBack(), null, new onProperty(), null);
+        do {
+            Thread.sleep(1000);
+        }
+        while (! Succeed.get());
+
+        log.debug("Subscribe to Desired properties on device Twin: PropertyKey={}...", TwinPropertyKey);
+        Map<Property, Pair<TwinPropertyCallBack, Object>> propertyPairMap =
+                singletonMap(new Property(TwinPropertyKey, null), new Pair<TwinPropertyCallBack, Object>(new onProperty(), null));
+        deviceClientManager.subscribeToTwinDesiredProperties(propertyPairMap);
+
+        log.debug("Start sending telemetry and reported property updates...");
+        new Thread(new Runnable() {
+
+            @Override
+            public void run() {
+                updateReportedProperties();
+                countDownLatch.countDown();
+            }
+        }).start();
+
+        new Thread(new Runnable() {
+
+            @Override
+            public void run() {
+                sendTelemetry();
+                countDownLatch.countDown();
+            }
+        }).start();
+
+        countDownLatch.await(MAX_APPLICATION_RUN_DURATION_IN_MINUTES, TimeUnit.MINUTES);
```

Comment:
does this sample need to demonstrate how to send telemetry and reported properties? I'd rather keep this sample simple. Open a connection, and respond to connection status updates

Reply:
I want to demonstrate that the device client can continue it's expected operations on reconnection. Since this is a reconnection sample, we want to cover the state of the device client in detail.

Reply:
Maybe talk with Priyanka about what she wants this sample to be. I had envisioned minimal code other than the connection status callback recommendations since it clutters up the sample. If Priyanka is okay with all this other code, then I'll sign off